src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,103 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private int currentCapacity;
+    private T[] listArray;
+    private int size;
+
+    public ArrayList() {
+        this.listArray = (T[]) new Object[DEFAULT_CAPACITY];
+        size = 0;
+        currentCapacity = DEFAULT_CAPACITY;
+    }
+
     @Override
     public void add(T value) {
-
+        add(value, size);
     }
 
     @Override
     public void add(T value, int index) {
+        if ((size == 0 && index == 0) || (index >= 0 && index <= size)) {
+            if (size == currentCapacity) {
+                currentCapacity = currentCapacity + (currentCapacity >> 1);
```

Comment:
make variable currentCapacity like local. 
Move implementation of listArray's resize to a separate method.(read checklist attendantly).
Your code has extra if-else block. try to decrease them.